src/analyzer/analyzermanager.cpp

```diff
@@ -0,0 +1,298 @@
+#include "analyzer/analyzermanager.h"
+
+#include <typeinfo>
+#include <QThread>
+
+#include <QtDebug>
+#include <QMutexLocker>
+#include <QListIterator>
+
+#include "library/trackcollection.h"
+#include "mixer/playerinfo.h"
+#include "track/track.h"
+#include "util/compatibility.h"
+#include "util/event.h"
+#include "util/timer.h"
+#include "util/trace.h"
+
+AnalyzerManager* AnalyzerManager::m_pAnalyzerManager = NULL;
+
+//static
+AnalyzerManager& AnalyzerManager::getInstance(UserSettingsPointer pConfig) {
+    if (!m_pAnalyzerManager) {
+        // There exists only one UserSettingsPointer in the app, so it doens't matter
+        // if we only assign it once.
+        m_pAnalyzerManager = new AnalyzerManager(pConfig);
+    }
+    int maxThreads = m_pAnalyzerManager->m_pConfig->getValue<int>(ConfigKey("[Library]", "MaxAnalysisThreads"));
+    int ideal = QThread::idealThreadCount();
```

Comment:
Oh. yup... Initially I was using the ideal in the if, instead of the hardcoded 32, but then i had the use case where i get -1 for ideal and i wasn't sure how to handle that.
Now that I think about it, i believe i will need to change this again, because now it does not work as I've explained above (it does not prevent opening 8 threads on a 4thread CPU)